src/fides/api/task/refine_target_path.py

```diff
@@ -116,6 +116,10 @@ def refine_target_path(
         next_levels = _enter_array(current_elem, target_path[1:], only)
         return _update_path(current_level, next_levels)
 
+    # stop recursion if current_elem is None
+    if current_elem is None:
+        return []
+
```

Comment:
This exit condition doesn't feel right - the target path wouldn't be built even if the current element legitimately pointed to None: 

For example, this is what I would expect here in a test:
```
    def test_refine_target_path_with_none_value(self):
        data = {"A": {"B": None}}
        assert refine_target_path(data, ["A", "B"]) == ["A", "B"]
```

Reply:
early thoughts, I'm wondering instead something like this, that is run after the "is the current_elem a list?" check.
```
 if len(target_path[1:]):
        next_levels = refine_target_path(current_elem, target_path[1:], only)
        return _update_path(current_level, next_levels)
```